src/Components/Endpoints/src/Builder/EndpointRouteBuilderResourceCollectionExtensions.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Endpoints;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// Extensions for <see cref="IEndpointConventionBuilder"/> to add resource collection metadata.
+/// </summary>
+public static class EndpointConventionBuilderResourceCollectionExtensions
+{
+    /// <summary>
+    /// Provides a helper to attach ResourceCollection metadata to endpoints.
+    /// </summary>
+    /// <param name="builder">The <see cref="IEndpointConventionBuilder"/>.</param>
+    /// <param name="manifestPath">The manifest associated with the assets.</param>
+    /// <returns>The <see cref="IEndpointConventionBuilder"/> that can be used to further configure the endpoints.</returns>
+    /// <remarks>
+    /// This method attaches static asset metadata to endpoints. It provides a simplified way to add 
+    /// resource collection metadata to any endpoint convention builder.
+    /// The <paramref name="manifestPath"/> must match the path of the manifest file provided to
+    /// the <see cref="StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets(IEndpointRouteBuilder, string?)"/> call.
+    /// </remarks>
+    public static IEndpointConventionBuilder WithStaticAssets(
+        this IEndpointConventionBuilder builder,
+        string? manifestPath = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        builder.Add(endpointBuilder =>
```

Comment:
Nit (only startup time, with a lot of endpoints):

Other builders are using some optimizations, e.g. https://github.com/dotnet/aspnetcore/blob/84747bb069d12ce059f09c8bd831741c6a74260d/src/Mvc/Mvc.ViewFeatures/src/Builder/ControllerActionEndpointConventionBuilderResourceCollectionExtensions.cs#L34

We don't have access to MVC's internals but we could try optimize it by moving the resolver creation and resolution outside the lambda.